ui/packages/consul-ui/lib/startup/templates/body.html.js

```diff
@@ -57,13 +59,18 @@ ${
 {{if .PartitionsEnabled}}
   <script data-app-name="${appName}" data-${appName}-routing src="${rootURL}assets/consul-partitions/routes.js"></script>
 {{end}}
+{{if .NamespacesEnabled}}
+  <script data-app-name="${appName}" data-${appName}-routing src="${rootURL}assets/consul-nspaces/routes.js"></script>
+{{end}}
 `
     : `
 <script>
 (
   function(get, obj) {
     Object.entries(obj).forEach(([key, value]) => {
-      if(get(key)) {
+      if(get(key) || (key === 'CONSUL_NSPACES_ENABLE' && ${
+        env('CONSUL_NSPACES_ENABLED') === '1' ? `true` : `false`
+      })) {
```

Comment:
Ok so this is the bit I'm not happy with but ok to punt for a bit. We've always had this awkward `ENABLE/ENABLED` thing happening. It started out as just `ENABLE`, but "reasons" happened. So then I made both work inside the codebase instead of trying to keep things consistent. Thing is there are a couple of sharp edges, i.e. in CI to enabled nspace testing the CI script uses "ENABLED", which is the name of a shell env var. Ideally the quick `env` function I wrote here (see above) would smooth that out - but...

Additionally this is going to be needed for testing partitions separately (and potentially ACLs separately, and potentially other things). So we need to add something here that is more generic (not just hardcode in the `NSPACES`). This is enough for the moment so I'll come back and revisit in a week or 2. All this is to do with testing/making it possible to test external feature flags, so it's not affecting anything that we ship.

When I come to do that revisit, I might stop smoothing over the `ENABLE/ENABLED` thing and just move everything to require `ENABLE` which I prefer (tell don't ask). It's getting to the point that every time I see this I wince.

Reply:
Yeah that's a really tough one to work with as the scope of the problem grows. We should definitely figure out how to push things toward consistency as this problem will only get worse as we go on.

Does this require coordination from the back-end to address properly/completely? I wonder if we could start on our end by using a function that can automatically `s/ENABLED/ENABLE` in env var names when they are first read into the startup script.

Reply:
> Does this require coordination from the back-end to address properly/completely?

🤔 Not entirely sure, but maybe it's not worth changing things up there. We have a place where we can draw the line between backend/frontend and do what works best for us (and backend can do whats best for them). I think the line is more blurred around CI things which is where the problem arose here, and a change there is super straightforwards that we can probably do ourselves.

Related: We are slowly but surely moving everything to use `(can "use nspaces")` etc instead of the env var directly. Thinking more into it I think there's something here around "setting means `ENABLE` (i.e. telling) and getting `ENABLED` (i.e. asking), our `can` helper is all about asking" - I'm kind talking out loud again here but it will maybe/probably sort itself out as we move everything over to `can` and become: `Please use `*_ENABLE` to configure environment variables, in the codebase please use `can` to inspect feature flags etc.` or something like that.

When we come to do this we can come back here and figure out what we want to do 👍